File: Stack/ArrayStack/arrayStack.cpp
#include <iostream>
using namespace std;

#define MAX_SIZE 100

class ArrayStack {
private:
    int top;
    int arr[MAX_SIZE] = {0};

public:
    ArrayStack() {
        top = -1;
    }

    void push() {
        if (top >= MAX_SIZE - 1) {
            cout<<"Stack is Full !!!";
            return;
        }
        int value = 0;
        cout<<"Enter the value to push >> ";
        cin>>value;
        arr[++top] = value;
    }
    
    void pop() {
        if (top < 0) {
            cout<<"Stack is Empty";
            return;
        }
        cout<<"Top Element Popped "<<arr[top];
        top--;
    }

    void peek() {
        if (top < 0) {
            cout<<"Stack is Empty";
            return;
        }
        cout<<"Stack Top is >> "<<arr[top];
    }

    void isEmpty() {
        if (top == -1) {
            cout<<"Stack is Empty !!!";
        } else {
            cout<<"Stack is not Empty !!!";
        }
    }

    void isFull() {
        if (top >= MAX_SIZE - 1) {
            cout<<"Stack is Full ";
        } else {
            cout<<"Stack is not Full !!!";
        }
    }

    void count() {
        cout<<"\nNumber of Element's in Stack is >> "<<top+1<<endl;
    }

    void destroy() {
        top = -1;
        cout<<"Stack Destroyed";
    }

    void display() {
        if (top < 0) {
            cout<<"\nStack is Empty !!";
            return;
        }
        for (int i = 0; i <= top; i++) {
            cout<<" ["<<arr[i];
        }
    }
};

int main() {
    int ch = 0;
    ArrayStack x;
    do {
        cout<<"\n\n1.Push\n2.Pop\n3.Top\n4.isEmpty\n5.isFull\n6.Count\n7.Destroy\n8.Display\n9.Exit \n>> ";
        cin >>ch;
        switch (ch) {
            case 1:
                x.push();
                break;
            case 2:
                x.pop();
                break;
            case 3:
                x.peek();
                break;
            case 4:
                x.isEmpty();
                break;
            case 5:
                x.isFull();
                break;
            case 6:
                x.count();
                break;
            case 7:
                x.destroy();
                break;
            case 8:
                x.display();
                break;
        }
    } while (ch != 9);
    return 0;
}

File: Stack/ArrayStack/arrayStack.py
class ArrayStack:
    MAX_SIZE = 100

    def __init__(self):
        self.top = -1
        self.arr = [0] * self.MAX_SIZE

    def push(self):
        if self.top >= self.MAX_SIZE - 1:
            print("Stack is Full !!!")
            return
        value = int(input("Enter the value to push >> "))
        self.top += 1
        self.arr[self.top] = value

    def pop(self):
        if self.top < 0:
            print("Stack is Empty")
            return
        print("Top Element Popped:", self.arr[self.top])
        self.top -= 1

    def peek(self):
        if self.top < 0:
            print("Stack is Empty")
            return
        print("Stack Top is >>", self.arr[self.top])

    def is_empty(self):
        if self.top == -1:
            print("Stack is Empty !!!")
        else:
            print("Stack is not Empty !!!")

    def is_full(self):
        if self.top >= self.MAX_SIZE - 1:
            print("Stack is Full ")
        else:
            print("Stack is not Full !!!")

    def count(self):
        print("\nNumber of Element's in Stack is >>", self.top + 1)

    def destroy(self):
        self.top = -1
        print("Stack Destroyed")

    def display(self):
        if self.top < 0:
            print("\nStack is Empty !!")
            return
        print("\nStack:", end="")
        for i in range(self.top + 1):
            print(" [", self.arr[i], end="")
        print("]")


def main():
    ch = 0
    x = ArrayStack()

    while ch != 9:
        print("\n\n1.Push\n2.Pop\n3.Top\n4.isEmpty\n5.isFull\n6.Count\n7.Destroy\n8.Display\n9.Exit \n>> ", end="")
        ch = int(input())
        if ch == 1:
            x.push()
        elif ch == 2:
            x.pop()
        elif ch == 3:
            x.peek()
        elif ch == 4:
            x.is_empty()
        elif ch == 5:
            x.is_full()
        elif ch == 6:
            x.count()
        elif ch == 7:
            x.destroy()
        elif ch == 8:
            x.display()


if __name__ == "__main__":
    main()